framework/topology.h

```diff
@@ -24,18 +24,11 @@ class LogicalProcessorSet;
 class Topology
 {
 public:
-
-    // Thread corresponds to the OS CPU.
-    struct Thread {
-        int id; // thread id within the core (e.g. 0 or 1 for SMT)
-        int cpu; // sandstone internal CPU id, e.g. tests get this identifier
-        int oscpu; // logical CPU id, as reported and recognized by the OS
-        Thread() noexcept : id(-1), cpu(-1), oscpu(-1) {}
-    };
+    using Thread = struct cpu_info;
```

Comment:
Fewer memory allocations and indirections. There's a common scenario that we'd only use `Thread` to access its `cpu` field and then get to `cpu_info`. This now avoids that extra indirection.

Previously, each `Core` was a `std::vector<Thread>` which means a memory allocation to store those three integers, which meant one memory allocation for each core in the system. It's now a `std::span<Thread>`, which points to existing memory instead of allocating more. The issue here is not the total amount of memory we allocated -- we still allocate memory proportional to the number of *cores* -- but the number of allocations. That is now only proportional to the number of *sockets*. This is an improvement because `malloc()` and family have a non-trivial runtime.

Though do note the sizes changed too, so we are in fact allocating less total memory:
| Type | Sizeof before | Sizeof after |
| --- | ---: | ---: |
| `Package` | 32 | 24 |
| `Core` | 32 | 24 |
| `Thread` | 12 | N/A |

Reply:
In summary:
- less indirection
- less operations to allocate memory
- less memory at the end

Ok, got it now. ty